doc/advanced-topics.md

```diff
@@ -418,6 +418,27 @@ For each attribute in the `reverse` array, an expression must be defined to calc
 notification attributes. This value will be passed to the underlying protocol with the `object_id` name. Details about
 how the value is then progressed to the device are protocol-specific.
 
+#### Autoprovision configuration (autoprovision)
```

Comment:
This PR adds three subsection at 4-level. The containing 3-level section is:

```
### Data mapping plugins
```

However, autoprovision/explicitAttr/appendMode are not relate with data mapping plugins.

Structure should be reviewed.

Reply:
Should we create a new section? In which case I don't know if we should group the 3 new descriptions under the same section or not.

Reply:
Not sure... I lack a global pespective of the topics in this file.

To know that, it would be great to do a multi-level ToC at the beginning of the file. Something like this: https://github.com/telefonicaid/fiware-cygnus/tree/master/doc/cygnus-ngsi/user_and_programmer_guide

Reply:
Added in https://github.com/telefonicaid/iotagent-node-lib/pull/1068/commits/9ecb677e15323c82b88a0b1dce2fc620287fa1ec

Reply:
After looking the ToC (thanks for put it together!) I don't see the new three topics fit in any existing group, so I'd suggest to add then at top-most level (maybe between "Metadata support" and "Data mapping plugins" topics).

Reply:
Done in https://github.com/telefonicaid/iotagent-node-lib/pull/1068/commits/99a7114e963e206a1b339f05f2bdf173cc7cef46